pghoard/basebackup.py

```diff
@@ -756,13 +824,46 @@ def run_local_tar_basebackup(self, delta=False):
                         pgdata=pgdata, tablespaces=tablespaces, target_chunk_size=target_chunk_size
                     )
                     chunks_count = len(chunks)
+
+                    delta_stats: Optional[Dict[str, int]] = None
+                    if with_delta_stats:
+                        delta_stats = {}
+
                     # Tar up the chunks and submit them for upload; note that we start from chunk 1 here; chunk 0
                     # is reserved for special files and metadata and will be generated last.
-                    chunk_files = self.create_and_upload_chunks(chunks, data_file_format, temp_base_dir)
+                    chunk_files = self.create_and_upload_chunks(
+                        chunks, data_file_format, temp_base_dir, delta_stats=delta_stats
+                    )
 
                     total_size_plain = sum(item["input_size"] for item in chunk_files)
                     total_size_enc = sum(item["result_size"] for item in chunk_files)
 
+                    if with_delta_stats:
+                        existing_hashes = self.fetch_all_data_files_hashes()
+                        new_hashes = {k: delta_stats[k] for k in set(delta_stats).difference(set(existing_hashes))}
+                        control_files_metadata_extra["delta_stats"] = {"new_hashes": new_hashes}
```

Comment:
Instead of storing new hashes here I think we need to store all hashes. Otherwise when oldest backup is removed we lose references to any hashes in it even though in actual delta backup mode the files would not be removed when they're still referenced by later backups.

Reply:
I mean all hashes that were contained in the backup itself. Storing all historical values too would soon balloon out of control.